test/TrueLayer.AcceptanceTests/PaymentTests.cs

```diff
@@ -48,6 +52,39 @@ public async Task Can_create_payment(CreatePaymentRequest paymentRequest)
             hppUri.ShouldNotBeNullOrWhiteSpace();
         }
 
+        [Fact]
+        public async Task Can_create_payment_with_auth_flow()
+        {
+            var sortCodeAccountNumber = new AccountIdentifier.SortCodeAccountNumber("567890", "12345678");
+            var providerSelection = new Provider.Preselected("mock-payments-gb-redirect", "faster_payments_service")
+            {
+                Remitter = new RemitterAccount("John Doe", sortCodeAccountNumber),
+            };
+            var paymentRequest = CreateTestPaymentRequest(
+                providerSelection,
+                sortCodeAccountNumber,
+                authorizationFlow: new StartAuthorizationFlowRequest(
+                    providerSelection, new SchemeSelection.Preselected(),
+                    new Redirect(new Uri("http://localhost:3000/callback")))
+            );
+
+            var response = await _fixture.Client.Payments.CreatePayment(
+                paymentRequest, idempotencyKey: Guid.NewGuid().ToString());
+
+            response.StatusCode.ShouldBe(HttpStatusCode.Created);
+            response.Data.IsT3.ShouldBeTrue();
+            CreatePaymentResponse.Authorizing authorizing = response.Data.AsT3;
+            authorizing.Id.ShouldNotBeNullOrWhiteSpace();
+            authorizing.ResourceToken.ShouldNotBeNullOrWhiteSpace();
+            authorizing.User.ShouldNotBeNull();
+            authorizing.User.Id.ShouldNotBeNullOrWhiteSpace();
+            authorizing.Status.ShouldBe("authorizing");
+
+            string hppUri = _fixture.Client.Payments.CreateHostedPaymentPageLink(
+                authorizing.Id, authorizing.ResourceToken, new Uri("https://redirect.mydomain.com"));
+            hppUri.ShouldNotBeNullOrWhiteSpace();
```

Comment:
Maybe we should check that the authorization_flow is not null and includes a redirect next action?

Reply:
done